datafusion/physical-expr/src/aggregate/count.rs

```diff
@@ -202,7 +202,9 @@ impl RowAccumulator for CountRowAccumulator {
     }
 
     fn evaluate(&self, accessor: &RowAccessor) -> Result<ScalarValue> {
-        Ok(accessor.get_as_scalar(&DataType::Int64, self.state_index))
+        Ok(ScalarValue::Int64(Some(
```

Comment:
is this change necessary or is it a drive by refactor?

Reply:
Drive-by mini bug fix if I am remembering correctly. I believe before this change, we were getting a NULL where we were supposed to get a zero in some use case. @mustafasrepo, am I remembering correctly? If so, we should add a mini-test for this.

Reply:
If the group, where `CountRowAccumulator` is working is empty. Without this change it produces `NULL`, However, it should produce `0`. I cannot produce an example for this case. Since row_accumulator only works when query contains `GROUP BY`, and `GROUP BY` cannot produce empty group. However, as an example 
``` sql
SELECT count(*) as cnt
FROM aggregate_test_100
WHERE 1 = 0;
```
will produce (where `CountAccumulator` is used)
```sql
"+-----+",
"| cnt |",
"+-----+",
"| 0   |",
"+-----+",
```
. However, if it were to work with `CountRowAccumulator`. It would produce
```sql
"+-----+",
"| cnt |",
"+-----+",
"|       |",
"+-----+",
```
which is wrong. This change fixes this bug. 
In summary, there is no way to reproduce this error (As far as I know) in current implementation, since `row_accumulator` is used only when query contains `GROUP BY` clause. However, if we were to use `row_accumulator` in non-group cases we can encounter this issue (I recognized this behavior when experimenting `RowAccumulator` support for non-group by aggregation). By the way, below query would 
``` sql
SELECT count(*)
FROM aggregate_test_100
WHERE 1 = 0
GROUP BY ();
```
reproduce the issue. However, it is not supported in datafusion currently.